engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,178 @@
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.platform.api.engine.ActionExecutionException;
+import org.pentaho.platform.api.engine.IAuthorizationPolicy;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ *  Provides a utility method to connect to repository and return the repository object
+ */
+public class RepositoryConnectionUtils {
+  private static final Log logger = LogFactory.getLog( RepositoryConnectionUtils.class );
+  private static final String SINGLE_DI_SERVER_INSTANCE = "singleDiServerInstance";
+  private static Class<?> PKG = RepositoryConnectionUtils.class;
+
+
+  /**
+   * Connects to the PDI repository. If the  system settings "SINGLE_DI_SERVER_INSTANCE" is set, it will connect to
+   * repository using the embedded repository.xml otherwise it will connect to repository with the given name.
+   *
+   * @param  String RepositoryName
+   * @param LoggingBuffer
+   * @return  Repository
+   * @throws KettleException
+   * @throws KettleSecurityException
+   * @throws ActionExecutionException
+   */
+  public static Repository connectToRepository( final String repositoryName, boolean isSingleDiServerInstance,
+             IAuthorizationPolicy authorizationPolicy, String userName,
+             String fullyQualifiedServerUrl, LoggingBuffer logBuffer ) throws KettleSecurityException, KettleException,
+              ActionExecutionException {
+
+    if ( !hasExecutePermission( authorizationPolicy ) ) {
+      throw new IllegalStateException(
+        BaseMessages.getString( "RepositoryConnectionUtils.ERROR_0001_NO_PERMISSION_TO_EXECUTE" ) ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( "Creating Meta-repository" );
+    }
+
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Populating Meta repository" );
+    }
+
+    try {
+      if ( isSingleDiServerInstance ) {
+        if ( logger.isDebugEnabled() ) {
+          logger.debug( "singleDiServerInstance=true, loading default repository" );
+        }
+
+        // Get the embedded repository xml
+        String repositoriesXml = getEmbeddedDefaultRepositoryXml( fullyQualifiedServerUrl );
+
+        ByteArrayInputStream sbis = new ByteArrayInputStream( repositoriesXml.getBytes( Const.XML_ENCODING ) );
+        repositoriesMeta.readDataFromInputStream( sbis );
+      } else {
+        // TODO: add support for specified repositories.xml files...
+        repositoriesMeta.readData(); // Read from the default $HOME/.kettle/repositories.xml file.
+      }
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString(
+        "RepositoryConnectionUtils.ERROR_0018_META_REPOSITORY_NOT_POPULATED" ), e ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Finding repository metadata" );
+    }
+    // Find the specified repository.
+    RepositoryMeta repositoryMeta = null;
```

Comment:
**Intellij warning:** Initializing this variable to null value is redundant

Reply:
It is surprising... My IntelliJ complains that the variables are not initialized

Reply:
Can be kept as is, just a warning.
But with the current code bellow it always gets initialized ( both **if** and **else** do it ). Only flow it doesn't get initialized is the catch case, but as the exception is thrown, the flow doesn't continue

Reply:
Attaching screenshot: <img width="605" alt="Screenshot 2019-04-15 at 17 51 37" src="https://user-images.githubusercontent.com/1265796/56150651-38490f00-5fa7-11e9-88d1-73890d649c9a.png">